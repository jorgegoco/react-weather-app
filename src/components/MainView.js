import React from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { BsArrowRightCircle } from 'react-icons/bs';
import getAllLocations from '../utils/mainData';

const MainView = () => {
  const [lat, lon] = useSelector((state) => state.coord.coords);
  const data = getAllLocations(lat, lon);
  const listItems = data.map((item) => (
    <div className="location-list" key={item[1]}>
      <Link to={`/detail/${item[0]}`}>
        <BsArrowRightCircle />
      </Link>
      <p>
        Location name:&nbsp;
        {item[0]}
      </p>
      <p>
        Location: latitude&nbsp;
        {item[1]}
        {' '}
        longitude&nbsp;
        {item[2]}
      </p>
      <p>
        Country:&nbsp;
        {item[3]}
      </p>
      <p>
        Population:&nbsp;
        {item[4]}
      </p>
    </div>
  ));
  return (
    <div className="locations-list">{listItems}</div>
  );
};

export default MainView;
