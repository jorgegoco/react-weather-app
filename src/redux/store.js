import { configureStore } from '@reduxjs/toolkit';
import coordReducer from './coordSlice';

const store = configureStore({
  reducer: {
    coord: coordReducer,
  },
});

export default store;
