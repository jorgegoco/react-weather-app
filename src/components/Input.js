import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { input } from '../redux/coordSlice';

const Input = () => {
  const [lat, lon] = useSelector((state) => state.coord.coords);
  const [value, setValue] = React.useState([lat, lon]);
  const dispatch = useDispatch();
  return (
    <div className="input-container">
      <h1>CHANGE ORIGIN :</h1>
      <input
        type="number"
        placeholder="Latitude..."
        value={value[0]}
        onChange={(e) => setValue([parseFloat(e.target.value), value[1]])}
        required
      />
      <input type="number" placeholder="Longitude..." value={value[1]} onChange={(e) => setValue([value[0], parseFloat(e.target.value)])} required />
      <button type="button" onClick={() => dispatch(input(value))}>Submit</button>
    </div>
  );
};

export default Input;
