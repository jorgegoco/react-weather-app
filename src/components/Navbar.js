import React from 'react';
import { Link } from 'react-router-dom';
import { FaChevronLeft, FaMicrophone } from 'react-icons/fa';
import { BsFillGearFill } from 'react-icons/bs';
import Input from './Input';

const Navbar = () => (
  <div className="navbar-container">
    <Link to="/">
      <FaChevronLeft className="chevron-icon" />
    </Link>
    <Input />
    <div className="navbar-subcontainer">
      <FaMicrophone />
      <BsFillGearFill />
    </div>
  </div>
);

export default Navbar;
