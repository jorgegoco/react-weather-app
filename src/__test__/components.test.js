import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';
import MainView from '../components/MainView';
import store from '../redux/store';
import Input from '../components/Input';

describe('Navbar-component renders', () => {
  test('It matchs the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Navbar />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Mainview-component renders', () => {
  test('It matchs the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <MainView />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Input-component renders', () => {
  test('It renders in the screen', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Input />
        </BrowserRouter>
      </Provider>,
    );

    expect(screen.getByText(/CHANGE/)).toBeInTheDocument();
  });

  test('It matchs the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Input />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
