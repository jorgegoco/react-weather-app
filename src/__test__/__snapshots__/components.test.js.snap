// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input-component renders It matchs the snapshot 1`] = `
<div
  className="input-container"
>
  <h1>
    CHANGE ORIGIN :
  </h1>
  <input
    onChange={[Function]}
    placeholder="Latitude..."
    required={true}
    type="number"
    value={43.467588}
  />
  <input
    onChange={[Function]}
    placeholder="Longitude..."
    required={true}
    type="number"
    value={-3.812048}
  />
  <button
    onClick={[Function]}
    type="button"
  >
    Submit
  </button>
</div>
`;

exports[`Mainview-component renders It matchs the snapshot 1`] = `
<div
  className="locations-list"
>
  <div
    className="location-list"
  >
    <a
      href="/detail/Santander"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <p>
      Location name: 
      Santander
    </p>
    <p>
      Location: latitude 
      43.464722
       
      longitude 
      -3.80444
    </p>
    <p>
      Country: 
      ES
    </p>
    <p>
      Population: 
      182700
    </p>
  </div>
  <div
    className="location-list"
  >
    <a
      href="/detail/El Astillero"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <p>
      Location name: 
      El Astillero
    </p>
    <p>
      Location: latitude 
      43.40094
       
      longitude 
      -3.82051
    </p>
    <p>
      Country: 
      ES
    </p>
    <p>
      Population: 
      17360
    </p>
  </div>
  <div
    className="location-list"
  >
    <a
      href="/detail/Camargo"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <p>
      Location name: 
      Camargo
    </p>
    <p>
      Location: latitude 
      43.40744
       
      longitude 
      -3.88498
    </p>
    <p>
      Country: 
      ES
    </p>
    <p>
      Population: 
      31404
    </p>
  </div>
  <div
    className="location-list"
  >
    <a
      href="/detail/Corvera"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <p>
      Location name: 
      Corvera
    </p>
    <p>
      Location: latitude 
      43.266972
       
      longitude 
      -3.9471
    </p>
    <p>
      Country: 
      ES
    </p>
    <p>
      Population: 
      15997
    </p>
  </div>
  <div
    className="location-list"
  >
    <a
      href="/detail/Torrelavega"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <p>
      Location name: 
      Torrelavega
    </p>
    <p>
      Location: latitude 
      43.349442
       
      longitude 
      -4.04785
    </p>
    <p>
      Country: 
      ES
    </p>
    <p>
      Population: 
      55947
    </p>
  </div>
  <div
    className="location-list"
  >
    <a
      href="/detail/Castro-Urdiales"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <p>
      Location name: 
      Castro-Urdiales
    </p>
    <p>
      Location: latitude 
      43.382858
       
      longitude 
      -3.22043
    </p>
    <p>
      Country: 
      ES
    </p>
    <p>
      Population: 
      31670
    </p>
  </div>
  <div
    className="location-list"
  >
    <a
      href="/detail/Getxo"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <p>
      Location name: 
      Getxo
    </p>
    <p>
      Location: latitude 
      43.356892
       
      longitude 
      -3.01146
    </p>
    <p>
      Country: 
      ES
    </p>
    <p>
      Population: 
      80770
    </p>
  </div>
  <div
    className="location-list"
  >
    <a
      href="/detail/Santurtzi"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <p>
      Location name: 
      Santurtzi
    </p>
    <p>
      Location: latitude 
      43.328419
       
      longitude 
      -3.03248
    </p>
    <p>
      Country: 
      ES
    </p>
    <p>
      Population: 
      46978
    </p>
  </div>
  <div
    className="location-list"
  >
    <a
      href="/detail/Algorta"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <p>
      Location name: 
      Algorta
    </p>
    <p>
      Location: latitude 
      43.34927
       
      longitude 
      -3.0094
    </p>
    <p>
      Country: 
      ES
    </p>
    <p>
      Population: 
      82624
    </p>
  </div>
</div>
`;

exports[`Navbar-component renders It matchs the snapshot 1`] = `
<div
  className="navbar-container"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <svg
      className="chevron-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
      />
    </svg>
  </a>
  <div
    className="input-container"
  >
    <h1>
      CHANGE ORIGIN :
    </h1>
    <input
      onChange={[Function]}
      placeholder="Latitude..."
      required={true}
      type="number"
      value={43.467588}
    />
    <input
      onChange={[Function]}
      placeholder="Longitude..."
      required={true}
      type="number"
      value={-3.812048}
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Submit
    </button>
  </div>
  <div
    className="navbar-subcontainer"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 352 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
      />
    </svg>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 1 0-5.86 2.929 2.929 0 0 1 0 5.858z"
      />
    </svg>
  </div>
</div>
`;
